nominatim/tools/collect_os_info.py

```diff
@@ -95,14 +95,18 @@ def report_system_information(config):
     """Generate a report about the host system including software versions, memory,
     storage, and database configuration."""
 
-    try:
-        with connect(config.get_libpq_dsn()) as conn:
-            postgresql_ver: str = convert_version(conn.server_version_tuple())
-            postgis_ver: str = convert_version(conn.postgis_version_tuple())
-    except DatabaseError:
-        with connect("dbname=postgres") as conn:
-            postgresql_ver: str = convert_version(conn.server_version_tuple())
-            postgis_ver: str = "Unable to connect to database"
+    with connect("dbname=postgres") as conn:
+        postgresql_ver: str = convert_version(conn.server_version_tuple())
+
+        with conn.cursor() as cur:
+            nominatim_db_exists = cur.execute("""
+            SELECT datname FROM pg_catalog.pg_database 
+            WHERE datname='nominatim'""")
```

Comment:
You cannot really make the assumption that the database is always called 'nominatim'. Users can use arbitrary names in their configuration.

That said, we cannot even make the assumption that they are using a local database. So I was wrong when recommending `connect("dbname=postgres")`. We need to use any other connection parameters from the user-supplied dsn. You can use [make_dsn()](https://www.psycopg.org/docs/extensions.html#psycopg2.extensions.make_dsn) to replace the database name in the DSN that you get from `config.get_libpq_dsn()`:

```
connect(psycopg2.make_dsn(config.get_libpq_dsn(), dbname='postgres'))
```

You can also use [parse_dsn](https://www.psycopg.org/docs/extensions.html#psycopg2.extensions.parse_dsn) to extract the database name from the DSN and then use the SELECT to find out if the database exists.